numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -922,8 +930,14 @@ get_c_wrapping_array(PyObject* NPY_UNUSED(self), PyObject* arg)
 
     flags = writeable ? NPY_ARRAY_WRITEABLE : 0;
     /* Create an empty array (which points to a random place) */
-    return PyArray_NewFromDescr(&PyArray_Type, PyArray_DescrFromType(NPY_INTP),
+    descr =  PyArray_DescrFromType(NPY_INTP);
+    Py_INCREF(descr);
+    ret = PyArray_NewFromDescr(&PyArray_Type, descr,
                                 1, &zero, NULL, &zero, flags, NULL);
+    if (ret == NULL)
+    {
+        Py_DECREF(descr);
+    }
```

Comment:
this fix came from reviewing the use of `PyArray_DescrFrom*` in the C file, and did not actually leak (yet).

Reply:
Wow, same bug yeah. Same thing, just make it `PyArray_DescrNewFromType`. With NULL check if you wish.

Reply:
fixed